libs/applePayments.js

```diff
@@ -61,15 +24,15 @@ api.processUser = function processUser (habitrpgUsers, user, jobStartDate, nextS
         for (let index in purchaseDataList) {
           let subscription = purchaseDataList[index];
           if (subscription.expirationDate > jobStartDate) {
-            return api.scheduleNextCheckForUser(habitrpgUsers, user, subscription, nextScheduledCheck);
+            return mobilePayments.scheduleNextCheckForUser(habitrpgUsers, user, subscription, nextScheduledCheck);
           }
         }
-        return api.cancelSubscriptionForUser(user);
+        return mobilePayments.cancelSubscriptionForUser(habitrpgUsers, user, "ios");
       }
     })
     .catch(err => {
-      if (err.validatedData && err.validatedData.is_retryable === false && err.validatedData.status === 21010) {
-        return api.cancelSubscriptionForUser(user);
+      if (err.status === 21010 ||  (err.validatedData && err.validatedData.is_retryable === false && err.validatedData.status === 21010)) {
```

Comment:
While we're tweaking the logic here, can we pull the `21010` magic number out to a constant? With a name (and maybe a comment, if necessary) that explains what it means.

Reply:
sounds good 👍